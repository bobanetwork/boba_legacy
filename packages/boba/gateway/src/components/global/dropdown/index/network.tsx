import React, { useState, useCallback } from 'react'
import ArrowDown from '../../../../images/icons/arrowdown.svg'
import {
  DropdownContainer,
  Header,
  IconContainer,
  Option,
  DefaultIcon,
  DropdownBody,
  Icon,
  DropdownContent,
} from '../themes/networkStyles'
interface IDropdownItem {
  value: string
  label: string
  imgSrc?: string
}

export interface IDropdownProps {
  items: IDropdownItem[]
  defaultItem: IDropdownItem
  onItemSelected?: (item: IDropdownItem) => void
}

export const Dropdown: React.FC<IDropdownProps> = ({
  items,
  defaultItem,
  onItemSelected,
}) => {
  const [selectedItem, setSelectedItem] = useState<IDropdownItem>(defaultItem)
  const [isOpen, setIsOpen] = useState<boolean>(false)

  const handleDropdown = useCallback(() => {
    setIsOpen(!isOpen)
  }, [isOpen])

  const selectItem = useCallback(
    (item: IDropdownItem) => {
      setSelectedItem(item)
      setIsOpen(false)
      onItemSelected && onItemSelected(item)
    },
    [onItemSelected]
  )

  return (
    <DropdownContainer className="dropdown">
      <Header onClick={handleDropdown}>
        <Option>
          {selectedItem.imgSrc && (
            <IconContainer>
              {selectedItem.imgSrc !== 'default' && (
                <Icon src={selectedItem.imgSrc} alt={selectedItem.label} />
              )}
              {selectedItem.imgSrc === 'default' && <DefaultIcon />}
            </IconContainer>
          )}
          {selectedItem.label}

          {isOpen ? <img src={ArrowDown} /> : <img src={ArrowDown} />}
        </Option>
      </Header>
      {isOpen && (
        <DropdownBody>
          <DropdownContent>
            {items.map((item, index) => (
              <Option
                key={index}
                onClick={() => selectItem(item)}
                style={{
                  flexDirection: 'row',
                  display: 'flex',
                  alignItems: 'center',
                  justifyContent: 'flex-start',
                }}
              >
                {item.imgSrc && (
                  <IconContainer style={{ alignContent: 'center' }}>
                    {item.imgSrc !== 'default' && (
                      <Icon
                        style={{ alignContent: 'center' }}
                        src={item.imgSrc}
                        alt={item.label}
                      />
                    )}
                    {item.imgSrc === 'default' && <DefaultIcon />}
                  </IconContainer>
                )}
                {item.label}
              </Option>
            ))}
          </DropdownContent>
        </DropdownBody>
      )}
    </DropdownContainer>
  )
}
