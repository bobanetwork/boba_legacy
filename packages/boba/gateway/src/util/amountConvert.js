/*
Copyright 2019-present OmiseGO Pte Ltd

Licensed under the Apache License, Version 2.0 (the "License");
you may not use this file except in compliance with the License.
You may obtain a copy of the License at

     http://www.apache.org/licenses/LICENSE-2.0

Unless required by applicable law or agreed to in writing, software
distributed under the License is distributed on an "AS IS" BASIS,
WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
See the License for the specific language governing permissions and
limitations under the License. */

// we use BigNumber here for decimal support
import BigNumber from 'bignumber.js';

export function logAmount (amount, power, truncate = 0) {

  const x = new BigNumber(amount);
  const exp = new BigNumber(10).pow(power);

  const calculated = x.div(exp);

  if(truncate > 0)
  	return calculated.toFixed(truncate);
  else
  	return calculated.toFixed();
}

/*Takes a value such as 3.92 and converts it into
a BigNumber in wei

Duplicates

ethers.utils.parseUnits( valueString , decimalsOrUnitName )   =>   BigNumber
*/

export function powAmount (amount, decimals) {

  const x = new BigNumber(amount)
  const exp = new BigNumber(10).pow(decimals)

  const calculated = x.multipliedBy(exp)
  return calculated.toFixed(0)
}

/* more clearly named version of this */
export function toWei_String(amount, decimals) {

  const x = new BigNumber(amount)
  const exp = new BigNumber(10).pow(decimals)

  const calculated = x.multipliedBy(exp)
  return calculated.toFixed(0)
}

export function amountToUsd(amount, lookupPrice, token) {
  if (token.symbol === 'ETH' && !!lookupPrice['ethereum']) {
    return amount * lookupPrice['ethereum'].usd
  } else if (token.symbol === 'BOBA' && !!lookupPrice[ 'boba-network' ]) {
    return amount * lookupPrice['boba-network'].usd
  } else if (token.symbol === 'OLO' && !!lookupPrice[ 'oolongswap' ]) {
    return amount * lookupPrice['oolongswap'].usd
  } else if (token.symbol === 'OMG' && !!lookupPrice[ 'omisego' ]) {
    return amount * lookupPrice['omisego'].usd
  } else if (token.symbol === 'USDC' && !!lookupPrice[ 'usd-coin' ]) {
    return amount * lookupPrice['usd-coin'].usd 
  } else if (!!lookupPrice[ token.symbol.toLowerCase() ]) {
    return amount * lookupPrice[token.symbol.toLowerCase()].usd
  } else {
    return 0
  }
}
