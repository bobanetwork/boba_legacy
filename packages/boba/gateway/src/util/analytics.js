import ReactGA from "react-ga4"
import { GA4_MEASUREMENT_ID } from "./constant";

export const init = () => {
  if (window.location.hostname === 'localhost') {
    return null;
  }
  if (GA4_MEASUREMENT_ID) {
    // init if the GA4 Measurement Id is available.
    ReactGA.initialize(GA4_MEASUREMENT_ID);
  }
}

export const sendPageView = (path) => {
  ReactGA.send({
    hitType: "pageview",
    page: path,
    title: path
  });
}


export default {
  init,
  sendPageView
}
