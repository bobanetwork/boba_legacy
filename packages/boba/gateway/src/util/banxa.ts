import { BANXA_URL } from './constant'

interface Props {
  symbol: string
  address: string
}

const FIAT_TYPE: string = 'USD'
const BLOCK_CHAIN: string = 'Boba Network'

export const prepareBanxaUrl = ({ symbol, address }: Props) => {
  const endpoint = BANXA_URL

  return `${endpoint}coinType=${symbol}&fiatType=${FIAT_TYPE}&blockchain=${BLOCK_CHAIN}&walletAddress=${address}`
}
