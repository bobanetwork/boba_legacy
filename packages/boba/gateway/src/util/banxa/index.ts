import axios from 'axios'

const BANXA_ENDPOINT = 'http://localhost:4000'

type FiatType = {
  fiat_code: string
  fiat_name: string
  fiat_symbol: string
}

type blockchainType = {
  code: string
  description: string
  is_default: boolean
}
type CoinType = {
  coin_code: string
  coin_name: string
  blockchains: blockchainType[]
}

type feesType = {
  name: string
  amount: number
  type: string
}
type TransactionFeesType = {
  fiat_code: string
  coin_code: string
  fees: feesType[]
}

type TransactionLimitType = {
  fiat_code: string
  min: number
  max: number
}
export type PaymentMethod = {
  id: number
  name: string
  description: string
  logo_url: string
  status: string
  type: string
  paymentType: string
  supported_agents: null
  supported_fiat: string[]
  supported_coin: string[]
  transaction_fees: TransactionFeesType[]
  transaction_limits: TransactionLimitType[]
}

export interface OrderParams {
  account_reference: string
  source: string
  source_amount: string | undefined
  target: string
  wallet_address: string
  return_url_on_success?: string
  payment_method_id: string
}

export const getStatics = async (): Promise<any> => {
  try {
    const response = await axios.get(`${BANXA_ENDPOINT}/statics`)
    return response.data
  } catch (error) {
    console.error(error)
  }
}

export const createOrder = (params: OrderParams) => {
  const parameters = {
    return_url_on_success: 'https://gateway.boba.network/?orderStatus=created',
    ...params,
  }

  axios
    .post(`${BANXA_ENDPOINT}/create-order`, parameters)
    .then((response) => {
      console.log(response.data)
    })
    .catch((error) => {
      console.error(error)
    })
}
