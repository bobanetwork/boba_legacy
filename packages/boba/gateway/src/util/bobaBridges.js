export const bobaBridges = [
  {
    "name": "Synapse",
    "type": "SYNAPSE",
    "link": "https://synapseprotocol.com/",
    "tokens": [
      "ETH",
      "nETH",
      "gOHM",
      "DAI",
      "USDC",
      "USDT",
      "SYN",
      "nUSD"
    ]
  },
  {
    "name": "Anyswap",
    "type": "ANYSWAP",
    "link": "https://anyswap.exchange/#/router",
    "tokens": [
      "MIM",
      "AVAX",
      "FRAX",
      "FTM",
      "FXS",
      "MATIC"
    ]
  },
  {
    "name": "Across",
    "type": "ACROSS",
    "link": "https://across.to/",
    "tokens": [
      "ETH",
      "BOBA",
      "USDC",
      "WBTC",
      "UMA"
    ]
  },
  {
    "name": "Celer",
    "type": "CELER",
    "link": "https://cbridge.celer.network/#/transfer",
    "tokens": [
      "ETH",
      "BOBA",
      "FRAX",
      "OLO"
    ]
  },
  {
    "name": "BoringDAO",
    "type": "BORINGDAO",
    "link": "https://oportal.boringdao.com/twoway",
    "tokens": [
      "USDT"
    ]
  },
  {
    "name": "PolyBridge",
    "type": "POLYBRIDGE",
    "link": "https://bridge.poly.network/",
    "tokens": [
      "BOBA"
    ]
  },
  {
    "name": "Symbiosis",
    "type": "SYMBIOSIS",
    "link": "https://app.symbiosis.finance/swap",
    "tokens": [
      "USDC"
    ]
  }
]
