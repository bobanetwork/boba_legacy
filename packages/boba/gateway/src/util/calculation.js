/*
  Varna - A Privacy-Preserving Marketplace
  Varna uses Fully Homomorphic Encryption to make markets fair. 
  Copyright (C) 2021 Enya Inc. Palo Alto, CA

  This program is free software: you can redistribute it and/or modify
  it under the terms of the GNU General Public License as published by
  the Free Software Foundation, either version 3 of the License, or
  (at your option) any later version.

  This program is distributed in the hope that it will be useful,
  but WITHOUT ANY WARRANTY; without even the implied warranty of
  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
  GNU General Public License for more details.

  You should have received a copy of the GNU General Public License
  along with this program. If not, see <https://www.gnu.org/licenses/>.
*/


const getCorrectResult = (type, num1, num2, result) => {
    const operations = {
        add: (num1, num2) => num1 + num2,
        sub: (num1, num2) => num1 - num2,
        div: (num1, num2) => num1 / num2,
        mul: (num1, num2) => num1 * num2
    };
    
    const temp_result = operations[type] ? operations[type](num1, num2) : result;
    return Math.abs(result - temp_result) > 1 ? temp_result : result;
};

const countDecimals = (num) => {
    const scientificRegex = /^-?\d+(?:\.\d+)?(?:[eE][+-]?\d+)?$/;
    const decimalRegex = /^-?\d+\.\d+$/;

    if (isNaN(num) || num === '') return 0; // Check if it's a valid number

    const str = num.toString(); // Convert number to String;


    // We check if number are cientific notation
    const scientificMatch = str.match(scientificRegex);
    if (scientificMatch) {
        const [, decimalPart, exponent] = scientificMatch[0].match(/^(-?\d+(?:\.\d+)?)(?:[eE]([+-]?\d+))?$/);
        return Math.max((decimalPart.split('.')[1] || '').length - (exponent || 0), 0);
    }

    // We check if number are decimal 
    const decimalMatch = str.match(decimalRegex);
    if (decimalMatch) {
        return (decimalMatch[0].split('.')[1] || '').length;
    }

    return 0;
}

const convertToInt = (num) => {
    const isExponential = num.toString().toUpperCase().split('E').length === 2;
    const decimalPlaces = isExponential ? countDecimals(num) : 0;
    const numWithoutDecimals = isExponential ? Math.round(num * Math.pow(10, decimalPlaces)) : num;
    return Number(numWithoutDecimals.toString().replace(".", ""));
}


var accAdd = function(num1, num2) {
    num1 = Number(num1);
    num2 = Number(num2);
    var dec1, dec2, times;
    try { dec1 = countDecimals(num1)+1; } catch (e) { dec1 = 0; }
    try { dec2 = countDecimals(num2)+1; } catch (e) { dec2 = 0; }
    times = Math.pow(10, Math.max(dec1, dec2));
    var result = (accMul(num1, times) + accMul(num2, times)) / times;
    return getCorrectResult("add", num1, num2, result);
};

var accSub = function(num1, num2) {
    num1 = Number(num1);
    num2 = Number(num2);
    var dec1, dec2, times;
    try { dec1 = countDecimals(num1)+1; } catch (e) { dec1 = 0; }
    try { dec2 = countDecimals(num2)+1; } catch (e) { dec2 = 0; }
    times = Math.pow(10, Math.max(dec1, dec2));
    var result = Number((accMul(num1, times) - accMul(num2, times)) / times);
    return getCorrectResult("sub", num1, num2, result);
};

var accDiv = function(num1, num2) {
    num1 = Number(num1);
    num2 = Number(num2);
    var t1 = 0, t2 = 0, dec1, dec2;
    try { t1 = countDecimals(num1); } catch (e) { }
    try { t2 = countDecimals(num2); } catch (e) { }
    dec1 = convertToInt(num1);
    dec2 = convertToInt(num2);
    var result = accMul((dec1 / dec2), Math.pow(10, t2 - t1));
    return getCorrectResult("div", num1, num2, result);
};

var accMul = function(num1, num2) {
    num1 = Number(num1);
    num2 = Number(num2);
    var times = 0, s1 = num1.toString(), s2 = num2.toString();
    try { times += countDecimals(s1); } catch (e) { }
    try { times += countDecimals(s2); } catch (e) { }
    var result = convertToInt(s1) * convertToInt(s2) / Math.pow(10, times);
    return getCorrectResult("mul", num1, num2, result);
};

export {
  accAdd,
  accSub,
  accDiv,
  accMul,
  countDecimals,
  convertToInt,
  getCorrectResult,
}