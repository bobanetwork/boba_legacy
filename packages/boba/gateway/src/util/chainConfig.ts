export const BobaChains = {
  // TODO: Consider using AddressManager or AddressPackage instead

  //#region boba_networks
  288: {
    url: 'https://replica.boba.network',
    testnet: false,
    name: 'Boba Ethereum Mainnet',
    teleportationAddress: '0xd68809330075C792C171C450B983F4D18128e9BF',
    height: 873302,
    supportedAssets: {
      ['0x0000000000000000000000000000000000000000'.toLowerCase()]: 'ETH',
      ['0xa18bF3994C0Cc6E3b63ac420308E5383f53120D7'.toLowerCase()]: 'BOBA',
      ['0x5DE1677344D3Cb0D7D465c10b72A8f60699C062d'.toLowerCase()]: 'USDT',
      ['0x68ac1623ACf9eB9F88b65B5F229fE3e2c0d5789e'.toLowerCase()]: 'BNB',
    },
  },
  56288: {
    url: 'https://replica.bnb.boba.network',
    testnet: false,
    name: 'Boba BNB Mainnet',
    teleportationAddress: '0xd68809330075C792C171C450B983F4D18128e9BF',
    height: 3393,
    supportedAssets: {
      ['0x4200000000000000000000000000000000000023'.toLowerCase()]: 'BNB',
      ['0x0000000000000000000000000000000000000000'.toLowerCase()]: 'BOBA',
    },
  },
  2888: {
    url: 'https://replica.goerli.boba.network',
    testnet: true,
    name: 'Boba Ethereum Goerli',
    teleportationAddress: '0xB43EE846Aa266228FeABaD1191D6cB2eD9808894',
    height: 40822,
    supportedAssets: {
      ['0x0000000000000000000000000000000000000000'.toLowerCase()]: 'ETH',
      ['0x4200000000000000000000000000000000000023'.toLowerCase()]: 'BOBA',
    },
  },
  9728: {
    url: 'https://replica.testnet.bnb.boba.network',
    testnet: true,
    name: 'Boba BNB Testnet',
    teleportationAddress: '0xf4d179d3a083Fa3Eede935FaF4C679D32d514186',
    height: 295353,
    supportedAssets: {
      ['0x4200000000000000000000000000000000000023'.toLowerCase()]: 'BNB',
      ['0x0000000000000000000000000000000000000000'.toLowerCase()]: 'BOBA',
      ['0xc614A66f82e71758Fa7735C91dAD1088c8362f15'.toLowerCase()]: 'ETH',
    },
  },
  //#endregion
  //#region l1
  1: {
    url: 'https://eth.llamarpc.com',
    testnet: false,
    name: 'Ethereum Mainnet',
    teleportationAddress: '0x0',
    height: 17565090,
    supportedAssets: {
      ['0x0000000000000000000000000000000000000000'.toLowerCase()]: 'ETH',
      ['0x42bBFa2e77757C645eeaAd1655E0911a7553Efbc'.toLowerCase()]: 'BOBA',
      ['0xdAC17F958D2ee523a2206206994597C13D831ec7'.toLowerCase()]: 'USDT',
      ['0xB8c77482e45F1F44dE1745F52C74426C631bDD52'.toLowerCase()]: 'BNB',
    },
  },
  56: {
    url: 'https://rpc.ankr.com/bsc',
    testnet: false,
    name: 'BNB Mainnet',
    teleportationAddress: '0x0',
    height: 30907682,
    supportedAssets: {
      /*'0x0000000000000000000000000000000000000000': 'ETH',
      '0x42bBFa2e77757C645eeaAd1655E0911a7553Efbc': 'BOBA',
      '0xdAC17F958D2ee523a2206206994597C13D831ec7': 'USDT',
      '0xB8c77482e45F1F44dE1745F52C74426C631bDD52': 'BNB',*/
    },
  },
  5: {
    url: 'https://goerli.gateway.tenderly.co',
    testnet: true,
    name: 'Goerli Testnet',
    teleportationAddress: '0x84b22166366a6f7E0cD0c3ce9998f2913Bf17A13',
    height: 9484025,
    supportedAssets: {
      ['0x0000000000000000000000000000000000000000'.toLowerCase()]: 'ETH',
      ['0xC2C527C0CACF457746Bd31B2a698Fe89de2b6d49'.toLowerCase()]: 'USDT',
      ['0xFC1C82c5EdeB51082CF30FDDb434D2cBDA1f6924'.toLowerCase()]: 'BNB',
      ['0xeCCD355862591CBB4bB7E7dD55072070ee3d0fC1'.toLowerCase()]: 'BOBA',
    },
  },
  97: {
    url: 'https://bsc.getblock.io/28c7f0fe-802f-4631-9f45-2df301789ecf/testnet/',
    testnet: true,
    name: 'BNB Testnet',
    teleportationAddress: '0x7f6a32bCaA70c65E08F2f221737612F6fC18347A',
    height: 32272487,
    supportedAssets: {
      ['0x0000000000000000000000000000000000000000'.toLowerCase()]: 'BNB',
      ['0x875cD11fDf085e0E11B0EE6b814b6d0b38fA554C'.toLowerCase()]: 'BOBA',
      ['0xd66c6B4F0be8CE5b39D52E0Fd1344c389929B378'.toLowerCase()]: 'ETH', // WETH
    },
  },
  //#endregion
}
