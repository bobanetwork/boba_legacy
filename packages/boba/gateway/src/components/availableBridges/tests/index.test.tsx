import { render, screen } from '@testing-library/react'
import React from 'react'
import { Provider } from 'react-redux'
import configureStore from 'redux-mock-store'
import CustomThemeProvider from 'themes'
import { AvailableBridges } from '..'
import { NETWORK, NETWORK_TYPE } from 'util/network/network.util'

const mockStore = configureStore()

const renderAvailableBridges = ({
  token,
  walletAddress,
  options = null,
}: any) => {
  return render(
    <Provider
      store={mockStore({
        ui: {
          theme: 'dark',
        },
        network: {
          activeNetwork: NETWORK.ETHEREUM,
          activeNetworkType: NETWORK_TYPE.MAINNET,
        },
        ...options,
      })}
    >
      <CustomThemeProvider>
        <AvailableBridges {...{ token, walletAddress }} />
      </CustomThemeProvider>
    </Provider>
  )
}

describe('Available Bridges', () => {
  let token = {
    symbol: 'ETH',
  }
  const walletAddress = 'random-code'

  describe('should match snapshot', () => {
    test('token value is null', () => {
      const { asFragment } = renderAvailableBridges({
        token: null,
        walletAddress,
      })
      expect(asFragment()).toMatchSnapshot()
    })
    test('network type is TESTNET', () => {
      const { asFragment } = renderAvailableBridges({
        token: null,
        walletAddress,
        options: {
          store: {
            network: {
              activeNetwork: NETWORK.ETHEREUM,
              activeNetworkType: NETWORK_TYPE.TESTNET,
            },
          },
        },
      })
      expect(asFragment()).toMatchSnapshot()
    })
    test('selected network is not ETHEREUM', () => {
      const { asFragment } = renderAvailableBridges({
        token: null,
        walletAddress,
        options: {
          store: {
            network: {
              activeNetwork: NETWORK.BNB,
              activeNetworkType: NETWORK_TYPE.TESTNET,
            },
          },
        },
      })
      expect(asFragment()).toMatchSnapshot()
    })

    test('connected on mainnet', () => {
      const { asFragment } = renderAvailableBridges({
        token,
        walletAddress,
      })
      expect(asFragment()).toMatchSnapshot()
    })
  })

  test('should see banxa for ETH token on mainnet', () => {
    renderAvailableBridges({
      token,
      walletAddress,
    })
    expect(screen.getByTestId('banxa')).toBeInTheDocument()
  })
  test('should see banxa for BOBA token on mainnet', () => {
    token = {
      symbol: 'BOBA',
    }
    renderAvailableBridges({
      token,
      walletAddress,
    })
    expect(screen.getByTestId('banxa')).toBeInTheDocument()
  })

  test('should not see banxa for other token on mainnet', () => {
    token = {
      symbol: 'USDC',
    }
    renderAvailableBridges({
      token,
      walletAddress,
    })
    expect(screen.queryByTestId('banxa')).not.toBeInTheDocument()
  })

  describe('Bridges should render correct for token', () => {
    test('should 2 bridge for ETH ', () => {
      renderAvailableBridges({
        token,
        walletAddress,
      })
      expect(screen.queryAllByTestId('bridge').length).toBe(2)
    })
    test('should 2 bridge for BOBA ', () => {
      renderAvailableBridges({
        token: { symbol: 'BOBA' },
        walletAddress,
      })
      expect(screen.queryAllByTestId('bridge').length).toBe(2)
    })
    test('should 2 bridge for USDC ', () => {
      renderAvailableBridges({
        token: { symbol: 'USDC' },
        walletAddress,
      })
      expect(screen.queryAllByTestId('bridge').length).toBe(2)
    })
  })
})
