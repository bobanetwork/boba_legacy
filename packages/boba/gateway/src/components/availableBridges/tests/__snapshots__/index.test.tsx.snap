// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Available Bridges should match snapshot connected on mainnet 1`] = `
<DocumentFragment>
  <div
    class="sc-ewnqHT govArP"
  >
    <p
      class="sc-bgqQcB sc-fFGjHI llaVrj jrSeWS"
    >
      Third party bridges
    </p>
    <a
      class="sc-dicizt dsNRSm"
      data-testid="banxa"
      href="https://boba.banxa.com/?coinType=ETH&fiatType=USD&blockchain=Boba Network&walletAddress=random-code"
      rel="noopener noreferrer"
      target="_blank"
    >
      <p
        class="sc-bgqQcB sc-iVCKna llaVrj bGQzVG"
      >
        Banxa
      </p>
    </a>
    <a
      class="sc-dicizt dsNRSm"
      data-testid="bridge"
      href="https://synapseprotocol.com/"
      rel="noopener noreferrer"
      target="_blank"
    >
      <p
        class="sc-bgqQcB sc-iVCKna llaVrj bGQzVG"
      >
        Synapse
      </p>
    </a>
    <a
      class="sc-dicizt dsNRSm"
      data-testid="bridge"
      href="https://cbridge.celer.network/#/transfer"
      rel="noopener noreferrer"
      target="_blank"
    >
      <p
        class="sc-bgqQcB sc-iVCKna llaVrj bGQzVG"
      >
        Celer
      </p>
    </a>
  </div>
</DocumentFragment>
`;

exports[`Available Bridges should match snapshot network type is TESTNET 1`] = `<DocumentFragment />`;

exports[`Available Bridges should match snapshot selected network is not ETHEREUM 1`] = `<DocumentFragment />`;

exports[`Available Bridges should match snapshot token value is null 1`] = `<DocumentFragment />`;
